lib/rdoc/parser/prism_ruby.rb

```diff
@@ -0,0 +1,1081 @@
+    a = RDoc::Alias.new(nil, old_name, new_name, comment, @singleton)
+    a.comment = comment
+    handle_modifier_directive(a, line_no)
+    a.store = @store
+    a.line = line_no
+    record_location(a)
+    if should_document?(a)
+      @container.add_alias(a)
+      @container.find_method(new_name, @singleton)&.visibility = visibility
+    end
+  end
+
+  # Handles `attr :a, :b`, `attr_reader :a, :b`, `attr_writer :a, :b` and `attr_accessor :a, :b`
+
+  def add_attributes(names, rw, line_no)
+    comment = consecutive_comment(line_no)
+    handle_consecutive_comment_directive(@container, comment)
+    return unless @container.document_children
+
+    names.each do |symbol|
+      a = RDoc::Attr.new(nil, symbol.to_s, rw, comment)
+      a.store = @store
+      a.line = line_no
+      a.singleton = @singleton
+      record_location(a)
+      handle_modifier_directive(a, line_no)
+      @container.add_attribute(a) if should_document?(a)
+      a.visibility = visibility # should set after adding to container
+    end
+  end
+
+  def add_includes_extends(names, rdoc_class, line_no) # :nodoc:
+    comment = consecutive_comment(line_no)
+    handle_consecutive_comment_directive(@container, comment)
+    names.each do |name|
+      ie = @container.add(rdoc_class, name, '')
+      ie.store = @store
+      ie.line = line_no
+      ie.comment = comment
+      record_location(ie)
+    end
+  end
+
+  # Handle `include Foo, Bar`
+
+  def add_includes(names, line_no) # :nodoc:
+    add_includes_extends(names, RDoc::Include, line_no)
+  end
+
+  # Handle `extend Foo, Bar`
+
+  def add_extends(names, line_no) # :nodoc:
+    add_includes_extends(names, RDoc::Extend, line_no)
+  end
+
+  # Adds a method defined by `def` syntax
+
+  def add_method(name, receiver_name:, receiver_fallback_type:, visibility:, singleton:, signature:, tokens:, start_line:, end_line:)
+    receiver = receiver_name ? find_or_create_module_path(receiver_name, receiver_fallback_type) : @container
+    meth = RDoc::AnyMethod.new(nil, name)
+    if (comment = consecutive_comment(start_line))
+      handle_consecutive_comment_directive(@container, comment)
+      handle_consecutive_comment_directive(meth, comment)
+
+      comment.normalize
+      comment.extract_call_seq(meth)
+      meth.comment = comment
+    end
+    handle_modifier_directive(meth, start_line)
+    handle_modifier_directive(meth, end_line)
+    return unless should_document?(meth)
+
+    meth.store = @store
+    meth.line = start_line
+
+    if meth.name == 'initialize' && !singleton
+      if meth.dont_rename_initialize
+        visibility = :protected
+      else
+        meth.name = 'new'
+        singleton = true
+        visibility = :public
+      end
+    end
+
+    meth.singleton = singleton
+    receiver.add_method(meth) # should add after setting singleton and before setting visibility
+    meth.visibility = visibility
+    if signature
+      meth.params ||= signature.params
```

Comment:
Let's avoid passing visitor objects around and simply pass `params`, `calls_super`, and `block_params` as separate arguments.

Reply:
done!